Sources/AsyncKit/ConnectionPool/EventLoopGroupConnectionPool.swift

```diff
@@ -100,8 +96,11 @@ public final class EventLoopGroupConnectionPool<Source> where Source: Connection
         on eventLoop: EventLoop? = nil,
         _ closure: @escaping (Source.Connection) -> EventLoopFuture<Result>
     ) -> EventLoopFuture<Result> {
-        self.pool(for: eventLoop ?? self.eventLoopGroup.next())
-            .withConnection(logger: logger ?? self.logger, closure)
+        guard !self.lock.withLock({ self.didShutdown }) else {
+            return (eventLoop ?? self.eventLoopGroup).future(error: ConnectionPoolError.shutdown)
+        }
+        return self.pool(for: eventLoop ?? self.eventLoopGroup.next())
+                   .withConnection(logger: logger ?? self.logger, closure)
```

Comment:
No "xcode" indentation please. 
```suggestion
           .withConnection(logger: logger ?? self.logger, closure)
```

Reply:
If you insist 😉